sentry-core/src/main/java/io/sentry/core/SentryClient.java

```diff
@@ -88,43 +88,83 @@ public SentryClient(final @NotNull SentryOptions options, @Nullable Connection c
                 SentryLevel.DEBUG,
                 "Event was dropped by processor: %s",
                 processor.getClass().getName());
+        sendEvent = false;
         break;
       }
     }
 
-    if (event == null) {
-      return SentryId.EMPTY_ID;
-    }
+    Session session = null;
 
-    // TODO: should it be done on the HUB?
-    updateSessionData(event, hint, scope);
+    if (event != null) {
+      session = updateSessionData(event, hint, scope);
 
-    if (!sample()) {
-      options
-          .getLogger()
-          .log(
-              SentryLevel.DEBUG,
-              "Event %s was dropped due to sampling decision.",
-              event.getEventId());
-      return SentryId.EMPTY_ID;
+      if (!sample()) {
+        options
+            .getLogger()
+            .log(
+                SentryLevel.DEBUG,
+                "Event %s was dropped due to sampling decision.",
+                event.getEventId());
+        sendEvent = false;
+        event = null;
```

Comment:
If I'm not missing something, if we're setting this to null here, we'll continue the execution and attempt to send the event.

Shouldn't we just return SentryId.Empty here?

Reply:
the event won't be sent, the flag is false and the event is now null, the session needs to be sent though, so the flow needs to continue